src/cascadia/WindowsTerminal/NonClientIslandWindow.cpp

```diff
@@ -267,13 +170,19 @@ LRESULT NonClientIslandWindow::HitTestNCA(POINT ptMouse) const noexcept
 // - A MARGINS struct containing the border dimensions we want.
 MARGINS NonClientIslandWindow::GetFrameMargins() const noexcept
 {
-    const auto titlebarView = GetTitlebarContentArea();
+    const auto dpi = ::GetDpiForWindow(_window);
+    const auto windowMarginSides = ::GetSystemMetricsForDpi(SM_CXDRAG, dpi);
+    const auto windowMarginBottom = ::GetSystemMetricsForDpi(SM_CXDRAG, dpi);
+
+    RECT buttonsRect = {};
+    ::DwmGetWindowAttribute(_window, DWMWA_CAPTION_BUTTON_BOUNDS, &buttonsRect, sizeof(buttonsRect));
+    const auto nonClientHeight = (buttonsRect.bottom - buttonsRect.top);

```

Comment:
I don't think this change to use the caption button height is right. I think this is what's causing the titlebar to have a bit of black space unter the caption buttons, but above the xaml island content.  I know we have work planned to draw the titlebar area with a color from the app, but I wouldn't want to complete the PR with the current black space there

Reply:
This is an inconsistency between the height of the Tab header and the height of the non client height. 
To resolve we can:
1. Tell Xaml about the actual non client height (but we don't want this as Xaml should be the source)
2. Pass the header height from Xaml to Non Client Window. 

I am going to implement 2.